lib/elixir/src/elixir_erl_try.erl

```diff
@@ -56,14 +56,12 @@ build_rescue(Meta, Parts, Body, S) ->
     elixir_erl_clauses:clause(Meta, fun elixir_erl_pass:translate_args/2,
                           Matches, Body, [], S),
 
-  TClauses =
-    [begin
-      TArgs   = [{tuple, Line, [{atom, Line, error}, TMatch, {var, Line, '_'}]}],
-      TGuards = elixir_erl_clauses:guards(Guards, [], TS),
-      {clause, Line, TArgs, TGuards, TBody}
-     end || {TMatch, {_, Guards}} <- lists:zip(TMatches, Parts)],
-
-  {TClauses, TS}.
+  Fun = fun({TMatch, {_, Guards}}, SG) ->
+    TArgs   = [{tuple, Line, [{atom, Line, error}, TMatch, {var, Line, '_'}]}],
+    {TGuards, ST} = elixir_erl_clauses:guards(Guards, [], SG),
+    {{clause, Line, TArgs, TGuards, TBody}, ST}
+  end,
+  lists:mapfoldl(Fun, TS, lists:zip(TMatches, Parts)).
```

Comment:
In fact this change is definitely breaking how try clauses work, as you are sharing the scope between clauses instead of starting a new one. Please revert. :)